src/Parser.zig

```diff
@@ -555,6 +555,80 @@ pub fn parse(self: *Parser, input: []const u8) !Result {
                     else => {},
                 }
             },
+            .osc => {
+                switch (b) {
+                    0x07, 0x1B => {
+                        state = .ground;
+                        if (b == 0x1b) {
+                            // advance one more for the backslash
+                            i += 1;
+                        }
+                        log.warn("unhandled osc: OSC {s}", .{input[start + 1 .. i + 1]});
+                        return .{
+                            .event = null,
+                            .n = i + 1,
+                        };
+                    },
+                    0x30...0x39 => {
+                        seq.param_buf[seq.param_buf_idx] = b;
+                        seq.param_buf_idx += 1;
+                    },
+                    ';' => {
+                        if (seq.param_buf_idx == 0) {
+                            seq.param_idx += 1;
+                        }
+                        if (seq.param_idx == 0) {
+                            const p = try std.fmt.parseUnsigned(u16, seq.param_buf[0..seq.param_buf_idx], 10);
+                            seq.param_buf_idx = 0;
+                            seq.param_idx += 1;
+                            switch (p) {
+                                52 => {
+                                    const allocator = self.grapheme_data.allocator;
```

Comment:
I think it would be best to pass an allocator to Tty.run (and Loop.run). The arg name should be like `paste_allocator` or something? This makes it a bit cleaner and lets people use a different allocator for this than the grapheme_data.

Reply:
Actually, maybe the vaxis struct should just hold onto the allocator it was passed on init and use that in `Tty.run`...what do you think?

Reply:
@neurocyte I was probably too late with the second comment before you ran off working on it! Sorry about that. What do you think about storing an allocator in the Vaxis struct to use for this? I think it might be cleaner that way - there are starting to be a lot of allocators passed around, but maybe that's preferred?

Reply:
I'm not so sure. I prefer to passing allocators explicitly to functions if at all possible. But I'm not using Loop.run or Tty.run so it's really up to you.